src/device/mod.rs

```diff
@@ -1306,8 +1304,36 @@ impl Runtime {
             // Refresh the lists of servers for STUN endpoint provider
             if let Some(direct) = self.entities.direct.as_ref() {
                 if let Some(stun_ep) = direct.stun_endpoint_provider.as_ref() {
+                    let use_ipv6 = self.features.ipv6 && {
```

Comment:
This causes on problem.

Now on every call to `telio_set_config()` new sockets will be allocated for `STUN`. This means, that new source port is allocated for STUN. While it is less relevant for WG-STUN requests, as those get encapsulated to the same WG tunnel anyways, the plaintext STUN requests will change its port (and thus `xor-mapped-address` attribute on every `telio-config` call. IMHO, changing ports here may cause problems for direct connections.